From: Syrone Wong <wong.syrone@gmail.com>
Date: Sat, 9 Apr 2022 13:35:00 +0800
Subject: [PATCH] luci-app-firewall: add fullcone4 / fullcone6 zone options

Add checkbox for fullcone4 and fullcone6 in zone edit page.
---
 applications/luci-app-firewall/luasrc/model/cbi/firewall/zone.lua | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/applications/luci-app-firewall/luasrc/model/cbi/firewall/zone.lua b/applications/luci-app-firewall/luasrc/model/cbi/firewall/zone.lua
index 102138c..4406050 100644
--- a/applications/luci-app-firewall/luasrc/model/cbi/firewall/zone.lua
+++ b/applications/luci-app-firewall/luasrc/model/cbi/firewall/zone.lua
@@ -181,6 +181,20 @@ o = s:option(Flag, "masq6", translate("IPv6 MASQ"))
 o.rmempty = false
 o:depends({ masq = "0" })

+o = s:option(Flag, "fullcone4", translate("IPv4 Fullcone NAT"))
+o.rmempty = false
+o.description = translate("Enable Fullcone NAT for IPv4 on this zone, replaces MASQ for fullcone ports. Requires kmod-nft-fullcone.")
+
+o = s:option(Flag, "fullcone6", translate("IPv6 Fullcone NAT"))
+o.rmempty = false
+o.description = translate("Enable Fullcone NAT for IPv6 on this zone. Usually not needed for IPv6.")
+
 o = s:option(Flag, "mtu_fix", translate("Force MSS clamping"))
 o.rmempty = false
 o.description = translate("Enable MSS clamping for TCP connections traversing this zone.")
--
2.35.3
